pkg/scalers/rabbitmq_scaler.go

```diff
@@ -494,3 +495,10 @@ func getMaximum(q []queueInfo) (int, float64) {
 	}
 	return maxMessages, maxRate
 }
+
+// Mask host for log porpouses
+func (s *rabbitMQScaler) anonimizeRabbitMQError(err error) error {
+	errorMessage := fmt.Sprintf("error inspecting rabbitMQ: %s", err)
+	m1 := regexp.MustCompile(`([^ \/:]+):([^\/:]+)\@`)
```

Comment:
You can move this out into a package level variable + init()

Reply:
which exactly are you talking about? The regex pattern?

Reply:
Yes:

```go
var anonymizePattern *regexp.Regexp

func init() {
  anonymizePattern = regex.MustCompile(...)
}
```

That means it only compiles the pattern once at startup.

Reply:
Nice improvement!!! 
I thought that golang cache several regex like other languages do (but I could do it wrong too, I'm learning golang right now)
I will update this with your suggestion asap :)

Reply:
Ahh, that explains why you're using a regex :) You could also do this with `net/url` `Parse` since it's a URI.

Reply:
oh, I got it! you are proposing to use something like this, right?
```golang
parsedURL, _ := url.Parse(s.metadata.host)
errorMessage := strings. ReplaceAllString(fmt.Sprintf("error inspecting rabbitMQ: %s", err),parsedURL.User.Password, "xxxx")
```

Don't take care about the exact code, I did manually in Github and surely it wrongs, but as idea

Reply:
What you have is probably better, just do the static precompile of the pattern and I think that's okay give or take if we want to ignore the username.

Reply:
👍

Reply:
Take a look when you can please :) 
I have applied your proposal :)